src/utp_stream.cpp

```diff
@@ -2640,7 +2640,7 @@ bool utp_socket_impl::incoming_packet(span<std::uint8_t const> buf
 		m_reply_micro = timestamp - ph->timestamp_microseconds;
 		std::uint32_t prev_base = m_their_delay_hist.initialized() ? m_their_delay_hist.base() : 0;
 		their_delay = m_their_delay_hist.add_sample(m_reply_micro, step);
-		int base_change = m_their_delay_hist.base() - prev_base;
+		int base_change = int(m_their_delay_hist.base()) - int(prev_base); // TODO: review this sign conversion
```

Comment:
here, I need some help. Below there is a line like `if (prev_base && base_change < 0 && base_change > -10000...`, but how `base_change` could be negative without overflow?